[issue] runtime error

**Describe the bug**
I updated my nuclei install to version `2.4.1` and now it errors out every time i try to run it. Be advised I think upgrade over brew install and i am running `Darwin HQSML-1689616 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64`. This is related to #888

**Nuclei version**
Please share the version of the nuclei you are running with `nuclei -version` 
See above and below

**Screenshot of the error or bug**
please add the screenshot showing bug or issue you are facing.
```
                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.4.1

		projectdiscovery.io

[ERR] Could not read nuclei-ignore file: open /Users/gbiago909/.config/nuclei/.nuclei-ignore: no such file or directory
[INF] Using Nuclei Engine 2.4.1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x49052fb]

goroutine 1 [running]:
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).RunEnumeration(0xc0000e6000, 0x0, 0x0)
	github.com/projectdiscovery/nuclei/v2/internal/runner/runner.go:345 +0xd5b
main.main()
	command-line-arguments/main.go:30 +0x87
```

I tried a specific URL and here is what i got
```
 nuclei -u https://redacted.example.com

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.4.3

		projectdiscovery.io

[ERR] Could not read nuclei-ignore file: open /Users/gbiago909/.config/nuclei/.nuclei-ignore: no such file or directory
[INF] Using Nuclei Engine 2.4.3
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x49ea03b]

goroutine 1 [running]:
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).RunEnumeration(0xc000156000, 0x0, 0x0)
	github.com/projectdiscovery/nuclei/v2/internal/runner/runner.go:361 +0xd5b
main.main()
	command-line-arguments/main.go:30 +0x87
```

So i tried building the package  from raw source and this is the err am receiving
```
go build -v -ldflags="-extldflags=-static" -o "nuclei" cmd/nuclei/main.go
github.com/projectdiscovery/gologger/levels
github.com/golang/groupcache/lru
golang.org/x/net/html/atom
golang.org/x/net/internal/iana
golang.org/x/sys/internal/unsafeheader
golang.org/x/text/encoding/internal/identifier
golang.org/x/text/internal/utf8internal
github.com/rivo/uniseg
github.com/go-rod/rod/lib/assets
golang.org/x/text/internal/tag
github.com/pkg/errors
github.com/cnf/structhash
github.com/projectdiscovery/stringsutil
github.com/projectdiscovery/gologger/writer
github.com/modern-go/concurrent
github.com/logrusorgru/aurora
github.com/blang/semver
gopkg.in/yaml.v2
github.com/google/go-querystring/query
github.com/apex/log
github.com/karrick/godirwalk
github.com/mattn/go-runewidth
github.com/projectdiscovery/hmap/store/cache
github.com/modern-go/reflect2
github.com/syndtr/goleveldb/leveldb/util
github.com/syndtr/goleveldb/leveldb/comparer
github.com/olekukonko/tablewriter
github.com/syndtr/goleveldb/leveldb/storage
github.com/golang/snappy
github.com/antchfx/xpath
github.com/syndtr/goleveldb/leveldb/cache
golang.org/x/net/html
github.com/syndtr/goleveldb/leveldb/filter
golang.org/x/text/transform
golang.org/x/text/internal/language
github.com/itchyny/timefmt-go
github.com/Knetic/govaluate
golang.org/x/text/encoding
github.com/google/go-github/github
github.com/syndtr/goleveldb/leveldb/opt
golang.org/x/text/encoding/internal
golang.org/x/text/runes
golang.org/x/text/encoding/charmap
github.com/itchyny/gojq
golang.org/x/text/encoding/japanese
github.com/syndtr/goleveldb/leveldb/errors
github.com/syndtr/goleveldb/leveldb/iterator
github.com/syndtr/goleveldb/leveldb/journal
github.com/syndtr/goleveldb/leveldb/memdb
github.com/syndtr/goleveldb/leveldb/table
golang.org/x/text/encoding/korean
golang.org/x/text/encoding/simplifiedchinese
github.com/syndtr/goleveldb/leveldb
golang.org/x/text/encoding/traditionalchinese
github.com/projectdiscovery/nuclei/v2/internal/colorizer
golang.org/x/text/encoding/unicode
golang.org/x/text/internal/language/compact
golang.org/x/text/language
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/helpers/deserialization
github.com/spaolacci/murmur3
git.mills.io/prologic/smtpd
golang.org/x/net/bpf
golang.org/x/sys/unix
github.com/eggsampler/acme/v3
github.com/jasonlvhit/gocron
github.com/google/uuid
github.com/projectdiscovery/goflags
github.com/karlseguin/ccache/v2
go.uber.org/atomic
github.com/json-iterator/go
github.com/projectdiscovery/clistats
github.com/projectdiscovery/nuclei/v2/pkg/types
github.com/projectdiscovery/nuclei/v2/pkg/operators/common/dsl
github.com/karlseguin/ccache
github.com/projectdiscovery/interactsh/pkg/storage
github.com/projectdiscovery/nuclei/v2/pkg/operators/matchers
golang.org/x/text/unicode/bidi
golang.org/x/text/unicode/norm
golang.org/x/net/http2/hpack
github.com/rs/xid
gopkg.in/corvus-ch/zbase32.v1
github.com/projectdiscovery/interactsh/pkg/server/acme
github.com/zclconf/go-cty/cty/set
golang.org/x/net/context/ctxhttp
github.com/hashicorp/go-cleanhttp
golang.org/x/time/rate
golang.org/x/oauth2/internal
github.com/hashicorp/go-retryablehttp
github.com/dgrijalva/jwt-go
github.com/fatih/structs
golang.org/x/text/encoding/htmlindex
golang.org/x/oauth2
golang.org/x/net/html/charset
golang.org/x/text/secure/bidirule
github.com/trivago/tgo/treflect
github.com/antchfx/htmlquery
go.uber.org/multierr
github.com/corpix/uarand
github.com/ysmood/gson
github.com/go-rod/rod/lib/js
github.com/ysmood/leakless/lib
github.com/trivago/tgo/tcontainer
github.com/ysmood/goob
github.com/dimchansky/utfbom
github.com/ysmood/leakless
github.com/segmentio/ksuid
github.com/valyala/bytebufferpool
github.com/go-rod/rod/lib/utils
github.com/go-rod/rod/lib/proto
github.com/andygrunwald/go-jira
github.com/go-rod/rod/lib/defaults
github.com/xanzy/go-gitlab
github.com/valyala/fasttemplate
github.com/andres-erbsen/clock
github.com/go-rod/rod/lib/cdp
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/replacer
go.uber.org/ratelimit
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/tostring
github.com/remeh/sizedwaitgroup
github.com/projectdiscovery/hmap/store/disk
golang.org/x/net/internal/socket
github.com/go-rod/rod/lib/launcher
github.com/shirou/gopsutil/v3/internal/common
github.com/tklauser/go-sysconf
golang.org/x/net/idna
github.com/projectdiscovery/hmap/store/hybrid
github.com/zclconf/go-cty/cty
github.com/projectdiscovery/nuclei/v2/pkg/projectfile
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/compare
github.com/spf13/cast
github.com/shirou/gopsutil/v3/cpu
github.com/shirou/gopsutil/v3/mem
golang.org/x/net/http/httpguts
golang.org/x/net/ipv4
golang.org/x/net/ipv6
golang.org/x/net/http2
github.com/shirou/gopsutil/v3/net
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/generators
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/expressions
github.com/projectdiscovery/rawhttp/client
golang.org/x/net/internal/socks
github.com/projectdiscovery/rawhttp/clientpipeline
golang.org/x/net/proxy
golang.org/x/net/publicsuffix
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/race
github.com/miekg/dns
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/raw
github.com/dsnet/compress/bzip2/internal/sais
github.com/dsnet/compress/internal
github.com/dsnet/compress/internal/errors
github.com/klauspost/cpuid
github.com/dsnet/compress
github.com/dsnet/compress/internal/prefix
github.com/klauspost/compress/flate
github.com/dsnet/compress/bzip2
github.com/owenrumney/go-sarif/sarif
github.com/ulikunitz/xz/internal/xlog
github.com/ulikunitz/xz/internal/hash
github.com/gosuri/uilive
github.com/ulikunitz/xz/lzma
github.com/klauspost/pgzip
github.com/gosuri/uiprogress/util/strutil
github.com/gosuri/uiprogress
github.com/tj/go-update/progress
github.com/projectdiscovery/gologger/formatter
github.com/projectdiscovery/gologger
github.com/projectdiscovery/nuclei/v2/pkg/catalog/config
github.com/projectdiscovery/nuclei/v2/pkg/progress
github.com/projectdiscovery/nuclei/v2/pkg/catalog
github.com/ulikunitz/xz
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader/filter
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader/load
github.com/c4milo/unpackit
github.com/tj/go-update
github.com/projectdiscovery/nuclei/v2/pkg/operators/extractors
github.com/projectdiscovery/nuclei/v2/pkg/operators
github.com/projectdiscovery/retryablehttp-go
github.com/projectdiscovery/rawhttp
github.com/tj/go-update/stores/github
github.com/shirou/gopsutil/v3/process
github.com/projectdiscovery/retryabledns
github.com/projectdiscovery/interactsh/pkg/server
github.com/projectdiscovery/nuclei/v2/pkg/protocols/dns/dnsclientpool
github.com/projectdiscovery/fastdialer/fastdialer
github.com/projectdiscovery/interactsh/pkg/client
github.com/projectdiscovery/nuclei/v2/pkg/output
github.com/go-rod/rod/lib/input
github.com/go-rod/rod/lib/devices
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolstate
github.com/projectdiscovery/nuclei/v2/pkg/protocols/network/networkclientpool
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/httpclientpool
github.com/projectdiscovery/nuclei/v2/pkg/reporting/dedupe
github.com/projectdiscovery/nuclei/v2/pkg/reporting/format
github.com/projectdiscovery/nuclei/v2/pkg/reporting/exporters/disk
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/github
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/gitlab
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/jira
github.com/projectdiscovery/nuclei/v2/pkg/reporting/exporters/sarif
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolinit
github.com/projectdiscovery/nuclei/v2/pkg/reporting
github.com/go-rod/rod
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh
github.com/projectdiscovery/nuclei/v2/pkg/protocols/headless/engine
github.com/projectdiscovery/nuclei/v2/pkg/protocols
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer
github.com/projectdiscovery/nuclei/v2/pkg/protocols/headless
github.com/projectdiscovery/nuclei/v2/pkg/workflows
github.com/projectdiscovery/nuclei/v2/pkg/protocols/dns
github.com/projectdiscovery/nuclei/v2/pkg/protocols/file
github.com/projectdiscovery/nuclei/v2/pkg/workflows/compile
github.com/projectdiscovery/nuclei/v2/pkg/protocols/offlinehttp
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http
github.com/projectdiscovery/nuclei/v2/pkg/protocols/network
github.com/projectdiscovery/nuclei/v2/pkg/templates
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/clusterer
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader
github.com/projectdiscovery/nuclei/v2/internal/runner
command-line-arguments
# command-line-arguments
/usr/local/Cellar/go/1.16.6/libexec/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
ld: library not found for -lcrt0.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)

make: *** [build] Error 2
```

here is what happened when i tried to update it
```
 sudo -EH go get -u -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei
Password:
go: downloading github.com/projectdiscovery/nuclei v1.1.7
go: downloading github.com/projectdiscovery/nuclei/v2 v2.4.3
go: downloading go.uber.org/atomic v1.9.0
go: downloading github.com/projectdiscovery/stringsutil v0.0.0-20210823090203-2f5f137e8e1d
go: downloading go.uber.org/multierr v1.7.0
go: downloading github.com/owenrumney/go-sarif v1.0.11
go: downloading github.com/segmentio/ksuid v1.0.4
go: downloading github.com/shirou/gopsutil/v3 v3.21.7
go: downloading github.com/shirou/gopsutil v3.21.7+incompatible
go: downloading github.com/mattn/go-runewidth v0.0.13
go: downloading github.com/google/go-querystring v1.1.0
go: downloading golang.org/x/net v0.0.0-20210825183410-e898025ed96a
go: downloading github.com/projectdiscovery/retryabledns v1.0.12
go: downloading github.com/projectdiscovery/fastdialer v0.0.12
go: downloading github.com/projectdiscovery/retryablehttp-go v1.0.2
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading github.com/andygrunwald/go-jira v1.14.0
go: downloading github.com/xanzy/go-gitlab v0.50.3
go: downloading github.com/ysmood/gson v0.7.0
go: downloading github.com/zclconf/go-cty v1.9.1
go: downloading github.com/antchfx/xpath v1.2.0
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/spf13/cast v1.4.1
go: downloading golang.org/x/sys v0.0.0-20210823070655-63515b42dcdf
go: downloading github.com/hashicorp/go-retryablehttp v0.7.0
go: downloading golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/klauspost/compress v1.13.4
go: downloading github.com/mattn/go-isatty v0.0.13
go: downloading github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d
go: downloading github.com/tklauser/go-sysconf v0.3.8
go: downloading github.com/StackExchange/wmi v1.2.1
go: downloading github.com/golang/snappy v0.0.4
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/go-ole/go-ole v1.2.4
go: downloading github.com/go-ole/go-ole v1.2.5
go: downloading github.com/tklauser/numcpus v0.2.1
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/tklauser/numcpus v0.3.0
go: downloading google.golang.org/protobuf v1.27.1
go: downloading github.com/eggsampler/acme v1.0.0
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/projectdiscovery/hmap v0.0.2-0.20210616215655-7b78e7f33d1f
go: downloading github.com/projectdiscovery/networkpolicy v0.0.1
go: downloading github.com/golang-jwt/jwt v3.2.1+incompatible
go: downloading github.com/golang-jwt/jwt v1.0.2
go: downloading github.com/akrylysov/pogreb v0.10.0
go: downloading github.com/dgraph-io/badger v1.6.2
go: downloading go.etcd.io/bbolt v1.3.5
go: downloading github.com/yl2chen/cidranger v1.0.2
go: downloading github.com/projectdiscovery/iputil v0.0.0-20210429152401-c18a5408ca46
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading github.com/projectdiscovery/iputil v0.0.0-20210804143329-3a30fcde43f3
go: downloading github.com/akrylysov/pogreb v0.10.1
go: downloading go.etcd.io/bbolt v1.3.6
go: downloading github.com/dgraph-io/ristretto v0.0.3
go: downloading github.com/projectdiscovery/mapcidr v0.0.6
go: downloading github.com/dgraph-io/ristretto v0.1.0
go: downloading github.com/projectdiscovery/mapcidr v0.0.8
go: downloading github.com/AndreasBriese/bbloom v0.0.0-20190825152654-46b345b51c96
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/projectdiscovery/blackrock v0.0.0-20210415162320-b38689ae3a2e
go: downloading github.com/golang/glog v1.0.0
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/karlseguin/ccache v1.0.1
go: downloading github.com/Knetic/govaluate v1.5.0
go: downloading github.com/tj/go-update v1.0.0
go: downloading github.com/blang/semver v1.1.0
github.com/dgraph-io/badger/options
google.golang.org/protobuf/internal/flags
golang.org/x/sys/internal/unsafeheader
google.golang.org/protobuf/internal/set
github.com/dgraph-io/badger/trie
github.com/golang/groupcache/lru
golang.org/x/net/html/atom
golang.org/x/text/encoding/internal/identifier
golang.org/x/text/internal/utf8internal
golang.org/x/net/internal/iana
github.com/projectdiscovery/blackrock
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/tostring
golang.org/x/text/internal/tag
github.com/projectdiscovery/hmap/store/cache
google.golang.org/protobuf/internal/pragma
github.com/projectdiscovery/stringsutil
github.com/google/go-querystring/query
github.com/mattn/go-runewidth
github.com/akrylysov/pogreb/fs
github.com/akrylysov/pogreb/internal/errors
github.com/akrylysov/pogreb/internal/hash
google.golang.org/protobuf/internal/detrand
google.golang.org/protobuf/internal/version
golang.org/x/net/internal/timeseries
golang.org/x/sys/unix
github.com/cespare/xxhash/v2
github.com/dgraph-io/ristretto/z/simd
github.com/projectdiscovery/goflags
github.com/olekukonko/tablewriter
github.com/dustin/go-humanize
github.com/google/go-github/github
github.com/golang/glog
google.golang.org/protobuf/internal/errors
github.com/AndreasBriese/bbloom
github.com/akrylysov/pogreb
golang.org/x/net/trace
github.com/golang/snappy
github.com/projectdiscovery/nuclei/v2/internal/colorizer
github.com/projectdiscovery/nuclei/v2/pkg/catalog
google.golang.org/protobuf/encoding/protowire
github.com/projectdiscovery/nuclei/v2/pkg/catalog/config
github.com/projectdiscovery/nuclei/v2/pkg/types
github.com/antchfx/xpath
golang.org/x/net/html
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader/filter
google.golang.org/protobuf/reflect/protoreflect
golang.org/x/text/transform
golang.org/x/text/internal/language
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/helpers/deserialization
golang.org/x/net/bpf
github.com/google/uuid
github.com/projectdiscovery/nuclei/v2/pkg/catalog/loader/load
golang.org/x/text/encoding
golang.org/x/text/runes
github.com/projectdiscovery/nuclei/v2/pkg/operators/common/dsl
golang.org/x/text/encoding/internal
github.com/projectdiscovery/interactsh/pkg/storage
go.uber.org/atomic
golang.org/x/text/unicode/bidi
github.com/syndtr/goleveldb/leveldb/table
golang.org/x/text/encoding/charmap
google.golang.org/protobuf/internal/encoding/messageset
google.golang.org/protobuf/internal/strs
google.golang.org/protobuf/internal/genid
google.golang.org/protobuf/internal/order
google.golang.org/protobuf/internal/encoding/text
google.golang.org/protobuf/reflect/protoregistry
google.golang.org/protobuf/runtime/protoiface
google.golang.org/protobuf/internal/descfmt
google.golang.org/protobuf/internal/descopts
golang.org/x/text/encoding/japanese
golang.org/x/text/encoding/korean
github.com/dgraph-io/badger/y
github.com/dgraph-io/ristretto/z
github.com/syndtr/goleveldb/leveldb
google.golang.org/protobuf/proto
go.etcd.io/bbolt
github.com/dgraph-io/badger/table
google.golang.org/protobuf/internal/encoding/defval
golang.org/x/text/encoding/simplifiedchinese
golang.org/x/text/encoding/traditionalchinese
golang.org/x/text/encoding/unicode
golang.org/x/text/internal/language/compact
golang.org/x/text/language
github.com/projectdiscovery/nuclei/v2/pkg/operators/matchers
google.golang.org/protobuf/encoding/prototext
google.golang.org/protobuf/internal/filedesc
github.com/dgraph-io/badger/skl
golang.org/x/net/internal/socket
github.com/projectdiscovery/clistats
golang.org/x/text/secure/bidirule
golang.org/x/text/unicode/norm
golang.org/x/net/http2/hpack
github.com/zclconf/go-cty/cty/set
github.com/projectdiscovery/nuclei/v2/pkg/progress
golang.org/x/net/context/ctxhttp
github.com/hashicorp/go-retryablehttp
golang.org/x/time/rate
github.com/golang-jwt/jwt
golang.org/x/oauth2/internal
go.uber.org/multierr
github.com/ysmood/gson
golang.org/x/net/ipv4
golang.org/x/net/ipv6
golang.org/x/oauth2
github.com/projectdiscovery/mapcidr
github.com/go-rod/rod/lib/utils
github.com/andygrunwald/go-jira
golang.org/x/text/encoding/htmlindex
github.com/xanzy/go-gitlab
github.com/go-rod/rod/lib/defaults
golang.org/x/net/html/charset
github.com/go-rod/rod/lib/proto
github.com/go-rod/rod/lib/cdp
github.com/projectdiscovery/iputil
github.com/yl2chen/cidranger/net
google.golang.org/protobuf/internal/encoding/tag
golang.org/x/net/idna
github.com/antchfx/htmlquery
github.com/zclconf/go-cty/cty
github.com/yl2chen/cidranger
github.com/go-rod/rod/lib/launcher
google.golang.org/protobuf/internal/impl
github.com/miekg/dns
github.com/projectdiscovery/nuclei/v2/pkg/operators/extractors
github.com/segmentio/ksuid
github.com/projectdiscovery/networkpolicy
github.com/projectdiscovery/nuclei/v2/pkg/operators
github.com/shirou/gopsutil/v3/internal/common
golang.org/x/net/http/httpguts
github.com/tklauser/go-sysconf
golang.org/x/net/http2
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/replacer
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/compare
github.com/spf13/cast
golang.org/x/net/internal/socks
golang.org/x/net/publicsuffix
github.com/shirou/gopsutil/v3/cpu
github.com/shirou/gopsutil/v3/mem
github.com/shirou/gopsutil/v3/net
golang.org/x/net/proxy
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/generators
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/race
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/raw
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/expressions
github.com/klauspost/compress/flate
github.com/owenrumney/go-sarif/sarif
github.com/klauspost/pgzip
github.com/c4milo/unpackit
github.com/tj/go-update
github.com/projectdiscovery/retryablehttp-go
github.com/projectdiscovery/rawhttp
github.com/tj/go-update/stores/github
google.golang.org/protobuf/internal/filetype
google.golang.org/protobuf/runtime/protoimpl
google.golang.org/protobuf/types/descriptorpb
github.com/projectdiscovery/retryabledns
github.com/projectdiscovery/interactsh/pkg/server
github.com/projectdiscovery/nuclei/v2/pkg/protocols/dns/dnsclientpool
github.com/projectdiscovery/interactsh/pkg/client
github.com/projectdiscovery/nuclei/v2/pkg/output
google.golang.org/protobuf/reflect/protodesc
github.com/projectdiscovery/nuclei/v2/pkg/reporting/dedupe
github.com/projectdiscovery/nuclei/v2/pkg/reporting/format
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/gitlab
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/jira
github.com/projectdiscovery/nuclei/v2/pkg/reporting/trackers/github
github.com/projectdiscovery/nuclei/v2/pkg/reporting/exporters/sarif
github.com/projectdiscovery/nuclei/v2/pkg/reporting/exporters/disk
github.com/projectdiscovery/nuclei/v2/pkg/reporting
github.com/golang/protobuf/proto
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/interactsh
github.com/go-rod/rod/lib/devices
github.com/go-rod/rod/lib/input
github.com/dgraph-io/badger/pb
github.com/dgraph-io/badger
github.com/go-rod/rod
github.com/shirou/gopsutil/v3/process
github.com/projectdiscovery/hmap/store/disk
github.com/projectdiscovery/hmap/store/hybrid
github.com/projectdiscovery/nuclei/v2/pkg/projectfile
github.com/projectdiscovery/fastdialer/fastdialer
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolstate
github.com/projectdiscovery/nuclei/v2/pkg/protocols/network/networkclientpool
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/httpclientpool
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolinit
github.com/projectdiscovery/nuclei/v2/pkg/protocols/headless/engine
# github.com/projectdiscovery/nuclei/v2/pkg/protocols/headless/engine
/Users/gbiago909/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.4.3/pkg/protocols/headless/engine/page_actions.go:469:27: too many arguments in call to page.Search
	have (number, number, string)
	want (string)
/Users/gbiago909/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.4.3/pkg/protocols/headless/engine/page_actions.go:473:9: invalid argument elms (type *rod.SearchResult) for len
/Users/gbiago909/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.4.3/pkg/protocols/headless/engine/page_actions.go:474:15: invalid operation: elms[0] (type *rod.SearchResult does not support indexing)
```

--- a/v2/internal/runner/runner.go
+++ b/v2/internal/runner/runner.go
@@ -379,8 +379,9 @@ func (r *Runner) RunEnumeration() error {
 	messageStr = builder.String()
 	builder.Reset()
 
-	gologger.Info().Msgf("Using Nuclei Templates %s%s", r.templatesConfig.CurrentVersion, messageStr)
-
+	if r.templatesConfig != nil {
+		gologger.Info().Msgf("Using Nuclei Templates %s%s", r.templatesConfig.CurrentVersion, messageStr)
+	}
 	if r.interactsh != nil {
 		gologger.Info().Msgf("Using Interactsh Server %s", r.options.InteractshURL)
 	}
@@ -513,6 +514,9 @@ func (r *Runner) RunEnumeration() error {
 
 // readNewTemplatesFile reads newly added templates from directory if it exists
 func (r *Runner) readNewTemplatesFile() ([]string, error) {
+	if r.templatesConfig == nil {
+		return nil, nil
+	}
 	additionsFile := filepath.Join(r.templatesConfig.TemplatesDirectory, ".new-additions")
 	file, err := os.Open(additionsFile)
 	if err != nil {
@@ -534,6 +538,9 @@ func (r *Runner) readNewTemplatesFile() ([]string, error) {
 
 // readNewTemplatesFile reads newly added templates from directory if it exists
 func (r *Runner) countNewTemplates() int {
+	if r.templatesConfig == nil {
+		return 0
+	}
 	additionsFile := filepath.Join(r.templatesConfig.TemplatesDirectory, ".new-additions")
 	file, err := os.Open(additionsFile)
 	if err != nil {

